Documentation/Proposals/NNNN-json-abi.md

```diff
@@ -0,0 +1,413 @@
+JSON objects representing each test that is part of the planned run. For
+example, this is the JSON representation of Swift Testing's own `canGetStdout()`
+test function:
+
+```json
+{
+  "kind": "test",
+  "payload": {
+    "displayName": "Can get stdout",
+    "id": "TestingTests.FileHandleTests/canGetStdout()/FileHandleTests.swift:33:4",
+    "isParameterized": false,
+    "kind": "function",
+    "name": "canGetStdout()",
+    "sourceLocation": {
+      "column": 4,
+      "fileID": "TestingTests/FileHandleTests.swift",
+      "line": 33
+    }
+  },
+  "version": 0
+}
+```
+
+A tool that is observing this data stream can build a map or dictionary of test
+IDs to comprehensive test details if needed. Once all tests in the planned run
+have been written out, testing begins. Swift Testing writes a sequence of JSON
+objects representing various events such as "test started" or "issue recorded".
+For example, here is an abridged sequence of events generated for a test that
+records a failed expectation:
+
+```json
+{
+  "kind": "event",
+  "payload": {
+    "instant": {
+      "absolute": 266418.545786299,
+      "since1970": 1718302639.76747
+    },
+    "kind": "testStarted",
+    "messages": [
+      {
+        "symbol": "default",
+        "text": "Test \"Can get stdout\" started."
+      }
+    ],
+    "testID": "TestingTests.FileHandleTests/canGetStdout()/FileHandleTests.swift:33:4"
+  },
+  "version": 0
+}
+
+{
+  "kind": "event",
+  "payload": {
+    "instant": {
+      "absolute": 266636.524236724,
+      "since1970": 1718302857.74857
+    },
+    "issue": {
+      "isKnown": false,
+      "sourceLocation": {
+        "column": 7,
+        "fileID": "TestingTests/FileHandleTests.swift",
+        "line": 29
+      }
+    },
+    "kind": "issueRecorded",
+    "messages": [
+      {
+        "symbol": "fail",
+        "text": "Expectation failed: (EOF → -1) == (feof(fileHandle) → 0)"
+      }
+    ],
+    "testID": "TestingTests.FileHandleTests/canGetStdout()/FileHandleTests.swift:33:4"
+  },
+  "version": 0
+}
+
+{
+  "kind": "event",
+  "payload": {
+    "instant": {
+      "absolute": 266636.524741106,
+      "since1970": 1718302857.74908
+    },
+    "kind": "testEnded",
+    "messages": [
+      {
+        "symbol": "fail",
+        "text": "Test \"Can get stdout\" failed after 0.001 seconds with 1 issue."
+      }
+    ],
+    "testID": "TestingTests.FileHandleTests/canGetStdout()/FileHandleTests.swift:33:4"
+  },
+  "version": 0
+}
+```
+
+Each event includes zero or more "messages" that Swift Testing intends to
+present to the user. These messages contain human-readable text as well as
+abstractly-specified symbols that correspond to the output written to the
+standard error stream of the test process. Tools can opt to present these
+messages in whatever ways are appropriate for their interfaces.
+
+### Invoking from the command line
+
+When invoking `swift test`, we propose adding three new arguments to Swift
+Package Manager:
```

Comment:
It may be best for this paragraph to be phrased in terms of _anticipated_ CLI flags that SwiftPM, or any similar tool which vends an executable entry point that wraps the call to the stable ABI entry point function, might vend.

Reply:
Sure, will rephrase.

Reply:
Added a paragraph indicating these are subject to change.